src/counts.jl

```diff
@@ -283,6 +283,40 @@ function addcounts_dict!(cm::Dict{T}, x::AbstractArray{T}) where T
     return cm
 end
 
+"""Specialist addcounts methods for small bits types"""
+function addcounts!(cm::Dict{Bool}, x::AbstractArray{Bool})
+    sumx = sum(x)
+    cm[true] = sumx
+    cm[false] = length(x) - sumx
+    cm
+end
+
+"""toindex functions to convert an integer to an array index between 1 to typemax{T}"""
```

Comment:
Same remark here. And "between 1 and typemax(T)".

Also better use the short syntax `toindex(x::Int16) = ...`, and use `typemax` for clarity.

Reply:
Actually, wouldn't it be enough to subtract `typemin(T)` and add `1` for all types? Then you wouldn't need these functions at all.

Reply:
would that slow things down? actually compiler might be small enough